### ISSUE TITLE
"as" in let destructure behaves strangely, also is not mentioned in the spec

### ISSUE THREAD

**Issue by ckp95** [austral/austral#382]
The following code compiles:

```
module body Mutate is
      record Other: Linear is
          y: Nat32;
      end;

      record Mut: Linear is
          x: Other;
      end;

      function main(): ExitCode is
          let foo: Mut := Mut(x => Other(y => 5));
          let { x: Other } := foo;
          let { y: Nat32 } := x;
          printLn(y);
          return ExitSuccess();
      end;
end module body.
```

But if I change the code to use `as` in the first let-destructure, it does not compile:

```
module body Mutate is
      record Other: Linear is
          y: Nat32;
      end;

      record Mut: Linear is
          x: Other;
      end;

      function main(): ExitCode is
          let foo: Mut := Mut(x => Other(y => 5));
          let { x as bar: Other } := foo;
          let { y: Nat32 } := bar;
          printLn(y);
          return ExitSuccess();
      end;
end module body.
```

Error message:

```
$ ./austral compile mine/mutate.aum --entrypoint=Mutate:main --output=mine/mutate
Compiler call tree printed to calltree.html
Error:
  Title: Generic Error
  Module:
    Mutate
  Location:
    [no span available]
  Description:
    Variable `x` not consumed at return statement.
  Code:
    [no span available]
```

However, this code using `as` *does* compile and works as expected:

```
module body Mutate is
      record Mut: Linear is
          x: Nat32;
      end;

      function main(): ExitCode is
          let foo: Mut := Mut(x => 5);
          let { x as bar: Nat32 } := foo;
          printLn(bar);
          return ExitSuccess();
      end;
end module body.
```

Also: I couldn't find where `as` is described in the spec. The let-destructure section doesn't mention it.

**Comment by eudoxia0** [comment]
Suppose this is related to #373

**Comment by eudoxia0** [comment]
This was fixed by #374 

I haven't added the destructuring explanation to the spec yet
